### DISCUSSION TITLE
MySQL이 요새 느리다는 신고가 들어왔습니다. 첫번째로 무엇을 확인하시고 조정하시겠나요?

### DISCUSSION THREAD

**Discussion by eatingbug** [Team-Psyduck/tech-sharing#76] [General]
.

**Comment by kkg5** [comment]
쿼리 프로파일링을 확인하거나 스프링 AOP로 속도를 측정해 어떤 쿼리가 오래 걸리는지 또는 자원을 많이 사용하는지를 확인하고 쿼리 튜닝을 먼저 진행할 것 같습니다.

**Comment by SionBackEnd** [comment]
mysql 기준 느린속도로 실행되는 query 찾아서 1초 이상 느려진 쿼리를 튜닝하는 방법이 있습니다. 
또한 프로세스를 확인하고 실행시간이 긴 프로세스를 확인 후 해당 프로세스를 종료 시키는 방법이 있습니다. 
```mysql
#느린 속도로 실행되는 query 찾기 
SELECT * FROM mysql.slow_log;

#process 확인하여 실행시간이 긴 프로세스 확인하기
SHOW FULL PROCESSLIST;
```

[참고](https://wisewiz.net/2017/08/03/mysql-db-%EB%8A%90%EB%A0%A4%EC%A1%8C%EC%9D%84%EB%95%8C-%ED%99%95%EC%9D%B8-%EB%B0%8F-%ED%95%B4%EA%B2%B0-%EB%B0%A9%EB%B2%95/)

**Comment by eatingbug** [comment]
MySQL DB Server 의 CPU, 메모리 등 리소스가 부족할 경우 성능저하가 발생할 수 있으므로, 충분히 확보되어있는지 확인한다.
또는 Slow Query Log 를 분석하여 성능저하를 일으키는 Query 를 개선하거나 MySQL 의 설정 (`innodb_buffer_pool_size`, `query_cache_size` ..) 를 적절하게 조정하여 성능을 개선한다.
